cognite_toolkit/cdf_tk/templates.py

```diff
@@ -447,293 +447,281 @@ def build_config(
 
     @property
     def added(self) -> list[ConfigEntry]:
-        return [entry for entry in self if entry.added]
+        return [entry for entry in self.values() if entry.is_added]
 
     @property
     def unchanged(self) -> list[ConfigEntry]:
-        return [entry for entry in self if entry.unchanged]
+        return [entry for entry in self.values() if entry.is_unchanged]
+
+    def dump(self) -> dict[str, Any]:
+        config: dict[str, Any] = {}
+        for entry in self.values():
+            local_config = config
+            for key in entry.key_path[:-1]:
+                if key not in local_config:
+                    local_config[key] = {}
+                local_config = local_config[key]
+            local_config[entry.key_path[-1]] = entry.value
+        config = self._reorder_config_yaml(config)
+        return config
+
+    def dump_yaml_with_comments(self, indent_size: int = 2) -> str:
+        """Dump a config dictionary to a yaml string"""
+        config = self.dump()
+        dumped = yaml.dump(config, sort_keys=False, indent=indent_size)
+        out_lines = []
+        if (entry := self.get(tuple())) and entry.comment:
+            for comment in entry.comment.above:
+                out_lines.append(f"# {comment}")
+        last_indent = 0
+        last_variable: str | None = None
+        path: tuple[str, ...] = tuple()
+        for line in dumped.splitlines():
+            indent = len(line) - len(line.lstrip())
+            if last_indent < indent:
+                if last_variable is None:
+                    raise ValueError("Unexpected state of last_variable being None")
+                path = (*path, last_variable)
+            elif last_indent > indent:
+                # Adding some extra space between modules
+                out_lines.append("")
+                indent_reduction_steps = (last_indent - indent) // indent_size
+                path = path[:-indent_reduction_steps]
+
+            variable = line.split(":", maxsplit=1)[0].strip()
+            if (entry := self.get((*path, variable))) and entry.comment:
+                for line_comment in entry.comment.above:
+                    out_lines.append(f"{' ' * indent}# {line_comment}")
+                if after := entry.comment.after:
+                    line = f"{line} # {after[0]}"
+
+            out_lines.append(line)
+            last_indent = indent
+            last_variable = variable
+        out_lines.append("")
+        return "\n".join(out_lines)
 
     def __str__(self) -> str:
         total_variables = len(self)
         lines = []
         if removed := self.removed:
-            lines.append(f"Removed {len(removed)} variables from config.yaml: {[str(r) for r in removed]}")
+            lines.append(f"Untracked {len(removed)} variables in config.yaml: {[str(r) for r in removed]}")
         if added := self.added:
             lines.append(f"Added {len(added)} variables to config.yaml: {[str(a) for a in added]}")
```

Comment:
This user experience isn't pleasent (you get a [var1, var2, ...] over multiple lines.

Reply:
You are right I have removed them. I have some extra print if the user adds --verbose.